### PR TITLE
[move][transactional tests] Add dev-inspect support

### PR DESCRIPTION
**Pull request by tnowacki**
## Description 

- Added minimal dev-inspect support for programmable transactions

## Test Plan 

- Added a new test. 
- I will use more extensively in #12055 

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

**Review by tzakian — APPROVED** [review]
LGTM!

**Review by amnn — APPROVED** [review]
LGTM!

File: crates/sui-core/src/authority/authority_test_utils.rs [hunk 1]
```diff
@@ -80,6 +80,9 @@ pub async fn send_and_confirm_transaction_with_execution_error(
 
     if with_shared {
         send_consensus(authority, &certificate).await;
+        if let Some(fullnode) = fullnode {
+            send_consensus(fullnode, &certificate).await;
```
[the remark this answers is not in the document]
Comment (tnowacki) [reply crates/sui-core/src/authority/authority_test_utils.rs:84]: Without this change, I got this error on all the shared object tests
```
test panicked: Shared object locks should have been set. tx_digset: ... 
```

[the remark this answers is not in the document]
Comment (tnowacki) [reply crates/sui-core/src/authority/authority_test_utils.rs:84]: I just don't think any of the authority tests that use full node with dev inspect used shared objects? 
Or I've done something else wrong and this just happened to fix it? Not sure

**Pull request merged**